openedx_learning/core/itemstore/models.py

```diff
@@ -0,0 +1,238 @@
+"""
+The item model hiearachy is: Item -> ItemVersion -> Content
+
+Item is the versionless thing that is guaranteed to exist for the lifetime of
+the LearningContext. An ItemVersion is a different version of that item for a
+given LearningContext, and may include policy changes (like grading). Content
+represents the raw byte data.
+"""
+from django.db import models
+from django.core.validators import MaxValueValidator
+
+from openedx_learning.lib.fields import (
+    hash_field,
+    identifier_field,
+    immutable_uuid_field,
+    manual_date_time_field,
+)
+from ..publishing.models import LearningContext, LearningContextVersion
+
+
+class Item(models.Model):
+    """
+    This represents any content that has ever existed in a LearningContext.
+
+    An Item will have many ItemVersions over time, and most metadata is
+    associated with the ItemVersion model. Make a foreign key to this model when
+    you need a stable reference that will exist for as long as the
+    LearningContext itself exists. It is possible for an Item to have no active
+    ItemVersion in the current LearningContextVersion (i.e. this content was at
+    some point removed from the "published" version).
+
+    An Item belongs to one and only one LearningContext.
+
+    The UUID should be treated as immutable. The identifier field *is* mutable,
+    but changing it will affect all ItemVersions.
+    """
+    uuid = immutable_uuid_field()
+    learning_context = models.ForeignKey(LearningContext, on_delete=models.CASCADE)
```

Comment:
So an item can only be in one `LearningContext`(LC), but since an `ItemVersion` can be in any number of `LearningContextVersions`(LCV) do we need a constraint somewhere that ensures that all those LCVs are from the same LC?  Or is that too paranoid at this stage?

Reply:
I like being paranoid about the data model in general, but I'm not sure what we can do with MySQL here, or how much overhead adding CheckConstraints is going to cause. I'll see how hard it is.